plugins/obs-webrtc/webrtc/src/output.rs

```diff
@@ -0,0 +1,385 @@
+use crate::whip;
+use anyhow::{anyhow, Result};
+use bytes::Bytes;
+use log::{debug, error, info, trace};
+use reqwest::Url;
+use std::boxed::Box;
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread::JoinHandle;
+use std::time::{Duration, Instant};
+use tokio::select;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::time::interval;
+use webrtc::api::interceptor_registry::register_default_interceptors;
+use webrtc::api::media_engine::{MediaEngine, MIME_TYPE_H264, MIME_TYPE_OPUS};
```

Comment:
I believe this (and a few other place) speaks to @lminiero's comment https://github.com/obsproject/obs-studio/pull/7926#issuecomment-1387254581 (tl;dr h264 is hardcoded in the rust code, so e.g., vp8 and vp9 will fail non-obviously).